### PR TITLE
drivers:platform:stm32:stm32_gpio.c Fix build for L5 family

### PR DESCRIPTION
**Pull request by rbolboac**
GPIO_MODE is defined in stm32l5xx_hal_gpio.c instead of stm32l5xx_hal_gpio.h so it has to be redefined in stm32_gpio.c before it is being used.
MODE_INPUT and MODE_OUTPUT are not defined in STM32 L5 hal, instead GPIO_MODE_INPUT and GPIO_MODE_OUTPUT_PP have to be used to obtain the same functionality.

### CODE REVIEW

File: drivers/platform/stm32/stm32_gpio.c
Comment (amiclaus) [inline drivers/platform/stm32/stm32_gpio.c:46]: nit: do we need brackets here?

**Review by amiclaus — APPROVED** [review]
lgtm, one minor nit.

File: drivers/platform/stm32/stm32_gpio.c
Comment (rbolboac) [reply drivers/platform/stm32/stm32_gpio.c:46]: I left the brackets there because I copied the definition directly from STM HAL.

Comment (amiclaus) [reply drivers/platform/stm32/stm32_gpio.c:46]: I guess it depends on the family: (see line 142) https://www.disca.upv.es/aperles/arm_cortex_m3/llibre/st/STM32F439xx_User_Manual/stm32f4xx__hal__gpio_8c_source.html

Comment (rbolboac) [reply drivers/platform/stm32/stm32_gpio.c]: I removed the brackets now. Let's wait for @buha 's approval before merging.

**Pull request merged**
